alerts/apps_alerts.libsonnet

```diff
@@ -235,11 +235,13 @@
             'for': '15m',
           },
           {
-            alert: 'KubeJobCompletion',
+            alert: 'KubeJobNotCompleted',
             expr: |||
-              kube_job_spec_completions{%(prefixedNamespaceSelector)s%(kubeStateMetricsSelector)s} - kube_job_status_succeeded{%(prefixedNamespaceSelector)s%(kubeStateMetricsSelector)s}  > 0
+              time() - max by(namespace, job_name) (kube_job_status_start_time{%(prefixedNamespaceSelector)s%(kubeStateMetricsSelector)s}
+                and
+              kube_job_status_active{%(prefixedNamespaceSelector)s%(kubeStateMetricsSelector)s} > 0) > 12 * 60 * 60
```

Comment:
With the `for` clause, it was easy for downstream users to patch the for duration but it's way harder in the new form. Would it make sense to have the threshold value as a jsonnet parameter?

Reply:
@simonpasquier, I'm thinking of the following expression along with `for` clause, as described [here](https://www.robustperception.io/alerting-on-gauges-in-prometheus-2-0), `avg_over_time` would mask the scrape failures. WDYT?

```
avg_over_time(kube_job_status_active[5m]) > 0
```

Reply:
last_over_time() would be better IMHO since it would resolve immediately after the job completes.
If the expression doesn't look at `kube_job_status_start_time` then the alert wouldn't fire until it has been evaluated for 12h?

Reply:
Actually I thought of reducing `for` value back [to 1hr](https://github.com/kubernetes-monitoring/kubernetes-mixin/pull/474). 12hr seem to bit high.